### PR TITLE
coverity related fixes (verbs, rxm, rxd, hook_hmem, sockets, util)

### PR DESCRIPTION
**Pull request by shefty**

### CODE REVIEW

File: prov/hook/hook_hmem/src/hook_hmem.c [hunk 1]
```diff
@@ -1663,6 +1663,12 @@ static int hook_hmem_mr_regv(struct fid *fid, const struct iovec *iov,
 	attr.auth_key_size = 0;
 	attr.auth_key = NULL;
 
+	/* hook_hmem_mr_regattr will determine the correct iface,
+	 * this silences warnings about uninitialized fields
+	 */
+	attr.iface = FI_HMEM_SYSTEM;
+	attr.device.reserved = 0;
```
Comment (rajachan) [inline prov/hook/hook_hmem/src/hook_hmem.c:1670]: We do require C99, so can initialize it up in L1655 as `struct fi_mr_attr attr = {0};` too.

File: README.md [hunk 2]
```diff
@@ -1,6 +1,5 @@
-[<img alt="libfabric master branch Travis CI status" src="https://travis-ci.org/ofiwg/libfabric.svg?branch=master"/>](https://travis-ci.org/ofiwg/libfabric)
 [<img alt="libfabric Coverity scan build status" src="https://scan.coverity.com/projects/4274/badge.svg"/>](https://scan.coverity.com/projects/4274)
```
Comment (rajachan) [inline README.md:1]: If you want a badge for the Github build check actions, you can use this:

```
[![Build Checks](https://github.com/ofiwg/libfabric/actions/workflows/pr-ci.yml/badge.svg)](https://github.com/ofiwg/libfabric/actions/workflows/pr-ci.yml)
```
Eg:
[Build Checks](https://github.com/ofiwg/libfabric/actions/workflows/pr-ci.yml)

File: prov/hook/hook_hmem/src/hook_hmem.c [hunk 3]
```diff
@@ -287,32 +291,39 @@ static int hook_hmem_track_atomic(struct hook_ep *ep, const struct fi_ioc *ioc,
...
+	}
 err3:
+	if (count)
+		hook_hmem_uncache_mr_iov(domain, iov, count);
+err2:
 	ofi_buf_free(*hmem_ctx);
-out:
-	ofi_mutex_lock(&domain->lock);
+err1:
+	ofi_mutex_unlock(&domain->lock);
```
Comment (rajachan) [inline prov/hook/hook_hmem/src/hook_hmem.c:329]: I was wondering how no one has tripped on this already when I saw this particular defect in the report.

**Review by rajachan — APPROVED** [review]

**Comment by rajachan** [comment]
This PR might address a couple of warnings from https://github.com/ofiwg/libfabric/issues/7376 too, so creating a link for reference.

**Review by j-xiong — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (shefty) [reply README.md:1]: Added via #7392 -- thanks!

**Comment by rajachan** [comment]
15 down:

```
    Analysis Summary:
       New defects found: 0
       Defects eliminated: 15

```

**Pull request merged**